RepresentsMe/MapViewController.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  Created by Jacob Hausmann on 2/22/19.
+//  Copyright © 2019 11foot8. All rights reserved.
+
+// code borrowed from tutorial at https://www.youtube.com/watch?v=2wxE8byc2FQ
+// https://www.dropbox.com/sh/keburc19mdiitj8/AADBqv7k0FnlWUQzIuni58Iba?dl=0
+//
+
+import UIKit
+import MapKit
+
+class MapViewController: UIViewController {
+    
+    let locationManager = CLLocationManager()
+    
+    // Regions will be 10 km across
+    let regionInMeters:CLLocationDistance = 10000
+    // Save previous location to limit geocode frequency
+    var previousLocation:CLLocation?
+    var previousGeocodeTime:Date?
+    // only request new geocodes when pin is moved 50+ meters
+    let minimumDistanceForNewGeocode:CLLocationDistance = 50
+    // only request new geocodes once every second
+    let minimumTimeForNewGecode:TimeInterval = 1
+    
+    let addressMessage = "Tap here to update address"
+
+    @IBOutlet weak var mapView: MKMapView!
+    @IBOutlet weak var addressButton: UIButton!
+    @IBOutlet weak var goButton: UIButton!
+    
+    @IBAction func addressButtonTouchUp(_ sender: Any) {
+        let center = getCenterLocation(for: mapView)
+        let time = Date()
+        guard let previousLocation = self.previousLocation else { return }
+        guard let previousGeocodeTime = self.previousGeocodeTime else { return }
+        
+        guard center.distance(from: previousLocation) > minimumDistanceForNewGeocode || time.timeIntervalSince(previousGeocodeTime) > minimumTimeForNewGecode else { return }
+        
+        getReverseGeocode()
+        
+        
+    }
+    @IBAction func goButtonTouchUp(_ sender: Any) {
+        // TODO: Check address validity
+        // TODO: Send address to representative table view
+    }
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        checkLocationServices()
+        // Do any additional setup after loading the view.
+    }
+    
+    func checkLocationServices() {
+        if CLLocationManager.locationServicesEnabled() {
+            // set up our location manager
+            setupLocationManager()
+            checkLocationAuthorization()
+        } else {
+            // show alert for letting user know they have to turn this on
+        }
+    }
+    
+    func setupLocationManager() {
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyBest
+    }
+    
+    func checkLocationAuthorization() {
+        switch CLLocationManager.authorizationStatus() {
+        case .authorizedWhenInUse:
+            startTrackingUserLocation()
+            break
+        case .denied:
+            // TODO: show alert instructing them how to turn on permissions
+            break
+        case .notDetermined:
+            locationManager.requestWhenInUseAuthorization()
+            break
+        case .restricted:
+            // TODO: show an alert letting them know whats up
+            break
+        case .authorizedAlways:
```

Comment:
Can you move this above the authorizedWhenInUse case, so that if either is set, user location will still be tracked?